src/library/scala/collection/IndexedSeqView.scala

```diff
@@ -96,23 +96,25 @@ object IndexedSeqView {
         r
       } else Iterator.empty.next()
 
-    override def drop(n: Int): Iterator[A] = {
-      if (n > 0) {
-        pos -= n
-        remainder = Math.max(0, remainder - n)
+    // from < 0 means don't move pos, until < 0 means don't limit remainder
+    //
+    override protected def sliceIterator(from: Int, until: Int): Iterator[A] = {
+      if (_hasNext) {
+        if (remainder <= from) remainder = 0                              // exhausted by big skip
+        else if (from <= 0) {                                             // no skip, pos is same
+          if (until >= 0 && until < remainder) remainder = until          // ...limited by until
+        }
+        else {
+          pos -= from                                                     // skip ahead
+          if (until >= 0 && until < remainder) {                          // ...limited by until
+            if (until <= from) remainder = 0                              // ...exhausted if limit is smaller than skip
+            else remainder = until - from                                 // ...limited by until, less the skip
+          }
+          else remainder -= from                                          // ...otherwise just less the skip
+        }
       }
       this
     }
-
-    override def sliceIterator(from: Int, until: Int): Iterator[A] = {
-      val startCutoff = pos
```

Comment:
I guess the wrong assumption here was that `pos` would be in the seq's range, but for an empty one it starts off at `-1`. Also the old implementation of `drop` could make it negative.

So it's good that the new implementation doesn't rely on `pos` in any way.

Reply:
Yes, I remember that part.